src/ray/gcs/redis_context.cc

```diff
@@ -171,6 +170,41 @@ RedisRequestContext::RedisRequestContext(instrumented_io_context &io_service,
   }
 }
 
+void RedisRequestContext::RedisResponseFn(struct redisAsyncContext *async_context,
```

Comment:
Nit: this doesn't need to be a member of `RedisRequestContext`?

Reply:
I tried removing it, but then it no longer has access to the private members of RedisRequestContext. But the static class method does have access.

Reply:
which private members it needs to access?

Reply:
`io_service_`, `exp_back_off_`, `redis_cmds_`, `pending_retries_`, `callback_`, `start_time_`